### PR TITLE
Minor changes to field codec functions

### PR DESCRIPTION
**Pull request by otoolep**
This small changes assist with distributed query support.

### CODE REVIEW

**Comment by otoolep** [comment]
@jwilder

File: tsdb/shard.go [hunk 1]
```diff
@@ -1081,6 +1025,16 @@ func (f *FieldCodec) DecodeByID(targetID uint8, b []byte) (interface{}, error) {
 	return 0, ErrFieldNotFound
 }
 
+// DecodeByName scans a byte slice for a field with the given name, converts it to its
+// expected type, and return that value.
+func (f *FieldCodec) DecodeByName(name string, b []byte) (interface{}, error) {
+	if fi := f.fieldByName(name); fi == nil {
+		return 0, fmt.Errorf("field doesn't exist")
```
Comment (jwilder) [inline tsdb/shard.go]: `ErrFieldNotFound`?

**Comment by jwilder** [comment]
One thing otherwise :+1:

**Comment by otoolep** [comment]
OK, thanks @jwilder -- I will make that change and merge.

**Pull request merged**